src/dmd/expressionsem.d

```diff
@@ -3587,36 +3587,58 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
         else if (tb.isscalar())
         {
+            Identifier id;
+            if (tb.isZeroInit(exp.loc) == true)
+            {
+                id = Identifier.idPool("___d_newitemT");
+            }
+            else
+            {
+                id = Identifier.idPool("___d_newitemiT");
+            }
```

Comment:
9 lines of code to one!
```
const id = Identifier.idPool(tb.isZeroInit(exp.log) ? "__d_newitemT" : __d_newitemiT");
```

Reply:
Now this is done in the `druntime` implementation of `___d_newitemT`